spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/WebMvcAutoConfiguration.java

```diff
@@ -506,16 +510,31 @@ private ResourceResolver getVersionResourceResolver(
 		private static final Log logger = LogFactory
 				.getLog(WelcomePageHandlerMapping.class);
 
-		private WelcomePageHandlerMapping(Resource welcomePage) {
-			if (welcomePage != null) {
+		private WelcomePageHandlerMapping(final ApplicationContext applicationContext,
+				final Resource welcomePage) {
+			if (welcomeTemplateExists(applicationContext)) {
```

Comment:
I abandoned the approach I took earlier in favor of the approach that uses the TemplateAvailabilityProviders, just as the Error templates do. However, what I noticed is that these TemplateAvailabilityProviders may say a template exists for one of the template engines simply because that template engine is on the classpath. 

For example, `classpath:/templates/example.ftl` may exist, and Freemarker might be on the classpath, but if Freemarker is not actually configured in the app context, this will erroneously say a view called "example" exists, when in fact nothing is configured to resolve that view. Maybe its not important as what I describe is an odd situation that people really shouldn't be doing.